pkg/kubebench/crd/writer.go

```diff
@@ -76,69 +49,24 @@ func (w *writer) Write(ctx context.Context, report starboard.CISKubeBenchOutput,
 		},
 		Report: report,
 	}, meta.CreateOptions{})
-	return
-}
-
-func (w *writer) getReportsByNodeName(ctx context.Context, name string) (reports []starboard.CISKubeBenchReport, err error) {
-	list, err := w.clientset.AquasecurityV1alpha1().CISKubeBenchReports().List(ctx, meta.ListOptions{
-		LabelSelector: labels.Set{
-			kube.LabelResourceKind: "Node",
-			kube.LabelResourceName: name,
-		}.String(),
-	})
-	if err != nil {
-		return
-	}
-	reports = list.Items
-	return
-}
-
-func (w *writer) removeHistoryLatestLabel(ctx context.Context, reports []starboard.CISKubeBenchReport) (err error) {
-	for _, report := range reports {
-		if value, ok := report.Labels[kube.LabelHistoryLatest]; !ok || value != "true" {
-			continue
-		}
-		clone := report.DeepCopy()
-		delete(clone.Labels, kube.LabelHistoryLatest)
-		klog.V(3).Infof("Removing %s label from %s report", kube.LabelHistoryLatest, clone.Name)
-		_, err = w.clientset.AquasecurityV1alpha1().CISKubeBenchReports().Update(ctx, clone, meta.UpdateOptions{})
+	if errors.IsAlreadyExists(err) {
```

Comment:
What if we always remove the existing resource before trying to create a new report or call `w.reports.Get` at first? I'm not sure I understand properly, but this report is generated for each node, right? In that case, creating a new report always fails after the second time. The current implementation calls 3 APIs if the report already exists, Create → Get → Update. I think we can call only 2  APIs, Delete → Create, or (Get → Create || Get → Update). The bad point of Delete → Create is that the report cannot be accessed only for a moment.

It may not make sense to reduce the number of API calls, but I think fewer is better than many. It also reduces source code.

Reply:
Yes. That's a very good point @knqyf263 I was thinking about that while implementing the change, but you convinced me that we might want to Get the report first because most of the time we'll be updating an existing report.

We don't want to Delete the report to handle the case when Create fails, which is quite likely in the distributed system

Reply:
>because most of the time we'll be updating an existing report.

Yes, that is exactly what I wanted to tell you. You understand what I meant better than me 😆 

>We don't want to Delete the report to handle the case when Create fails, which is quite likely in the distributed system

Right. The system which is able to accept the case sometimes goes with the way for simplicity. In this case, it is probably better to call `Get` first.

Reply:
Okey. I've changed the code to Get and then either Create (if error was not found) or Update (if the report was there)